buildSrc/src/main/kotlin/io/micronaut/build/LicenseExtension.kt

```diff
@@ -19,6 +19,7 @@ abstract class LicenseExtension(@Inject val initScript: RegularFile) {
             this.repoDirectory.set(checkoutDir)
             this.branch.set(branch)
             this.uri.set(uri)
+            this.cleanupGitRepo.set("true".equals(project.findProperty("cleanupGitRepo") ?: "false"))
```

Comment:
I think it's a mistake since you're passing a `String` to a property which wants a `Boolean`. Should be:

```java
this.cleanupGitRepo.set(providers.gradleProperty("cleanupGitRepo").map(Boolean::parseBoolean).getOrElse(false))
```

Reply:
Changed to use `ProviderFactory`